webapp/src/modules/parcels/sagas.js

```diff
@@ -37,6 +46,61 @@ function* handleParcelsRequest(action) {
   }
 }
 
+function* handleParcelRequest(action) {
+  try {
+    const { x, y } = action
+    const nw = buildCoordinate(x, y)
+    const se = buildCoordinate(x, y)
+
+    let [parcels, data] = yield all([
+      api.fetchParcels(nw, se),
+      api.fetchParcelData(x, y)
+    ])
+
+    parcels = parcels.map(parcel => Object.assign(parcel, { data }))
+
+    yield put({
+      type: FETCH_PARCEL_SUCCESS,
+      parcels
+    })
+  } catch (error) {
+    console.warn(error)
+    yield put({
+      type: FETCH_PARCEL_FAILURE,
+      error: error.message
+    })
+  }
+}
+
+function* handleParcelDataRequest(action) {
+  try {
+    const { x, y } = action
+    if (!inBounds(x, y)) {
+      throw new Error(`Coords (${x}, ${y}) are outside of the valid bounds`)
+    }
+
+    const parcels = yield select(getParcels)
+    const parcel = parcels[buildCoordinate(x, y)]
+    if (!parcel) {
+      throw new Error(`Parcel (${x}, ${y}) is not in the state:`, parcels)
```

Comment:
Careful here! as far as I know, the second argument on the Error constructor is ignored. Only the first `message` argument is used

![image](https://user-images.githubusercontent.com/692077/35799331-a6bdcf0e-0a65-11e8-87f6-062a61d7f1b1.png)